dockstore-client/src/main/java/io/github/collaboratory/cwl/CWLClient.java

```diff
@@ -632,8 +624,13 @@ private String cleanFileId(String fileId) {
         return inputSet;
     }
 
+    private String getString(Map map, String key) {
+        Object value = map.get(key);
+        return value instanceof String ? (String)value : null;
+    }
+
     private String getPathOrLocation(Map param) {
-        return ObjectUtils.firstNonNull((String)param.get("path"), (String)param.get("location"));
+        return ObjectUtils.firstNonNull(getString(param, "path"), getString(param, "location"));
```

Comment:
(Feel free to ignore) If these keys are referenced elsewhere, it might be nice to split them out into static variables, i.e. `DESCRIPTOR_PATH_KEY_ONE` and `DESCRIPTOR_PATH_KEY_TWO`. But, if they are only used here then that may be overkill.